updates/restart.go

```diff
@@ -0,0 +1,48 @@
+package updates
+
+import (
+	"context"
+	"time"
+
+	"github.com/safing/portbase/log"
+	"github.com/safing/portbase/modules"
+	"github.com/tevino/abool"
+)
+
+const (
+	// RestartExitCode will instruct portmaster-start to restart the process immediately, potentially with a new version.
+	RestartExitCode = 23
+)
+
+var (
+	restartTask      *modules.Task
+	restartPending   *abool.AtomicBool
+	restartTriggered *abool.AtomicBool
+)
+
+func init() {
+	restartTask = module.NewTask("automatic restart", automaticRestart)
+}
+
+func triggerRestart(delay time.Duration) {
+	restartPending.Set()
+	restartTask.Schedule(time.Now().Add(delay))
+}
```

Comment:
What about exporting this so we can use in in core/events.go instead of using `updates.RestartExitCode` and `modules.Shutdown()` directly. 

```suggestion
// DelayedRestart triggers a restart of the application by shutting down the module system
// and returning with RestartExitCode. If a restart is already pending this is a no-op.
func DelayedRestart(delay time.Duration) {
    if restartPending.SetToIf(false, true) {
         restartTask.Schedule(time.Now().Add(delay))
    }
}
```

Reply:
`DelayedRestart` is meant to be able to update an existing delay. I updated the function description.